library/std/src/sys/wasi/alloc.rs

```diff
@@ -1,42 +1,66 @@
+#![deny(unsafe_op_in_unsafe_fn)]
+
 use crate::alloc::{GlobalAlloc, Layout, System};
 use crate::ptr;
 use crate::sys_common::alloc::{realloc_fallback, MIN_ALIGN};
 
+// SAFETY: All methods implemented follow the contract rules defined
+// in `GlobalAlloc` trait
```

Comment:
should the FIXME below be mentioned here, like it was not satisfied?

Reply:
Yes, please do so.